src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -64,10 +54,8 @@ public override long Position
 
             public override long Seek(long offset, SeekOrigin origin)
             {
-                CheckDisposed();
+                CheckCanSeek();
```

Comment:
Note that calling SeekAsync on this stream will now throw, whereas it should call into the sync version. It's important to carefully go over all possible such behavioral changes, and possibly add tests to make sure everything works correctly.

Reply:
Fixed. Moved to async implementation.